sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewTestSuite.scala

```diff
@@ -865,6 +865,168 @@ class PersistedViewTestSuite extends SQLViewTestSuite with SharedSparkSession {
+          "c1 type should be updated to bigint")
+        assert(c2Final.nonEmpty && c2Final(0).getString(1) == "double",
+          "c2 type should be updated to double")
+        assert(c1Final.nonEmpty && c1Final(0).getString(2) == "user comment 1",
+          "c1 comment should still be 'user comment 1' (preserved)")
+        assert(c2Final.nonEmpty && c2Final(0).getString(2) == "user comment 2",
+          "c2 comment should still be 'user comment 2' (preserved)")
+      }
+    }
+  }
+
+  test("Schema evolution comments legacy behavior with preserveUserComments=false") {
+    withSQLConf(VIEW_SCHEMA_EVOLUTION_PRESERVE_USER_COMMENTS.key -> "false") {
+      withTable("t") {
+        withView("v") {
+          // Create table with comments.
+          sql("CREATE TABLE t (c1 INT COMMENT " +
+            "'table comment 1', c2 STRING COMMENT 'table comment 2')")
+          sql("INSERT INTO t VALUES (1, 'a'), (2, 'b')")
+
+          // Create view with schema evolution.
+          sql("CREATE VIEW v WITH SCHEMA EVOLUTION AS SELECT * FROM t")
+
+          // User manually changes view comments.
+          val catalog = spark.sessionState.catalog
+          val viewMeta = catalog.getTableMetadata(TableIdentifier("v"))
+          val newSchema = StructType(Seq(
+            StructField("c1", IntegerType, nullable = true).withComment("user comment 1"),
+            StructField("c2", StringType, nullable = true).withComment("user comment 2")
+          ))
+          catalog.alterTable(viewMeta.copy(schema = newSchema))
+
+          // Verify manually set comments.
+          val descManual = sql("DESCRIBE EXTENDED v").collect()
+          val c1CommentManual = descManual.filter(r => r.getString(0) == "c1")
+          val c2CommentManual = descManual.filter(r => r.getString(0) == "c2")
+          assert(c1CommentManual.nonEmpty && c1CommentManual(0).getString(2) == "user comment 1")
+          assert(c2CommentManual.nonEmpty && c2CommentManual(0).getString(2) == "user comment 2")
+
+          // SELECT from view (triggers ViewSyncSchemaToMetaStore).
+          checkAnswer(sql("SELECT * FROM v"), Seq(Row(1, "a"), Row(2, "b")))
+
+          // With flag=false, comments should REVERT to table comments (legacy behavior).
+          val descAfterSelect = sql("DESCRIBE EXTENDED v").collect()
+          val c1CommentAfter = descAfterSelect.filter(r => r.getString(0) == "c1")
+          val c2CommentAfter = descAfterSelect.filter(r => r.getString(0) == "c2")
+          assert(c1CommentAfter.nonEmpty && c1CommentAfter(0).getString(2) == "table comment 1",
+            "c1 comment should revert to 'table comment 1' (legacy behavior)")
+          assert(c2CommentAfter.nonEmpty && c2CommentAfter(0).getString(2) == "table comment 2",
+            "c2 comment should revert to 'table comment 2' (legacy behavior)")
+        }
+      }
+    }
+  }
+
+  test("View schema evolution config controls comment preservation behavior") {
```

Comment:
is this test a duplication of the previous two tests which test config on and off behaviors?